src/c4/test/tstC4_Status.cc

```diff
@@ -0,0 +1,95 @@
+//----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   c4/test/tstC4_Status.cc
+ * \author Robert B. Lowrie
+ * \date   Friday May 26 19:58:19 2017
+ * \brief  Unit test for C4_Status class.
+ * \note   Copyright (C) 2016-2017 Los Alamos National Security, LLC.
+ *         All rights reserved.
+ */
+//---------------------------------------------------------------------------//
+
+#include "c4/ParallelUnitTest.hh"
+#include "ds++/Release.hh"
+#include <vector>
+
+using namespace std;
+using rtt_c4::C4_Status;
+using rtt_c4::C4_Req;
+
+//---------------------------------------------------------------------------//
+// TESTS
+//---------------------------------------------------------------------------//
+
+void tst2Procs(rtt_dsxx::UnitTest &ut) {
+
+  if (rtt_c4::nodes() != 2)
+    return;
+
+  PASSMSG("Running tst2Procs.");
+
+  const int my_proc = rtt_c4::node();
+  C4_Status status;
+  C4_Req request;
+
+  const int num_int = 2;
+  const int num_double = 5;
+  const int tag = 101;
+
+  if (my_proc == 0) {
+    std::vector<int> send_buffer(num_int);
+    std::vector<double> recv_buffer(num_double);
+    rtt_c4::receive_async(request, &recv_buffer[0], num_double, 1, tag);
```

Comment:
yeah, good idea.  It does check status.get_source(), but this is a further check.